Snakefile

```diff
@@ -0,0 +1,116 @@
+import itertools
+
+HEADINGS = ["botany", "developmental_biology", "genetics","microbiology", "ecology",
+            # Informatics subheadings
+            "cheminformatics", "computational_biology", "consumer_health_informatics","medical_informatics",
+            # Algorithms subheadings
+            "artificial_intelligence", "latent_class_analysis",
+            ]
+
+COCI_DIR = '/mnt/SlowData/coci'
+
+SPLIT_HEADINGS = [h1 + '-' + h2 for h1, h2 in itertools.combinations(sorted(HEADINGS), 2)]
+SPLIT_HEADINGS2 = [h2 + '-' + h1 for h1, h2 in itertools.combinations(sorted(HEADINGS), 2)]
+
+wildcard_constraints:
+    # Random seeds should be numbers
+    shuffle="\d+",
+    # The headings wildcard used in shuffle_networks can contain letters, underscores,
+    # and dashes, but no other characters such as numbers
+    heading="[a-z_-]+",
+    heading1="[a-z_-]+",
+    heading2="[a-z_-]+",
+
+ruleorder: split_combined_shuffled_networks > shuffle_networks
```

Comment:
Why do you need `ruleorder` here? I thought in Snakemake the order rules run in was dictated by the input and output files that need to be generated - does it not automatically handle this case?

Reply:
As far as I can tell, Snakemake looks at a target file from `rule all` and says "which rule can create that output?" For whatever reason it is convinced that `shuffle_networks` can create files from `split_combined_shuffled_networks`, despite the fact that their outputs are in different directories. I think it tries to coerce the `heading` wildcard to eat some of the directory portion of the output file, despite the fact that slashes are explicitly disallowed :shrug:

In general, the AmbiguousRuleExceptions are weird and somewhat broken and I try to just find what fixes them and not think too hard about the underlying software https://github.com/snakemake/snakemake/issues/674